src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp

```diff
@@ -81,38 +176,53 @@ show_device_conf(std::shared_ptr<xrt_core::device>& dev)
 static void 
 update_daemon_config(const std::string& host)
 {
-  std::cout << "update config with " << host;
+  XBU::sudo_or_throw("Updating daemon configuration requires sudo");
```

Comment:
xbmgmt can't run any commands without sudo. Is this needed?

Reply:
sudo is not enforced by xbmgmt2, don't know if it should be? But certain operations that change the device are not possible unless run as sudo, rather than mysteriously erroring out sudo_or_throw will cause graceful exit with proper message.

Reply:
Hmm. I think we do enforce sudo:
```
/opt/xilinx/xrt/bin/xbmgmt --new status
-----------------------------------------------------
 Invoking next generation of the xbmgmt application
-----------------------------------------------------
XRT build version: 2.8.0
Build hash: 9aa1c2cd35d8ad401c673a9ea6d5630ede456f99
Build date: 2020-08-14 10:31:27
Git branch: master
PID: 219041
UID: 31939
[Tue Aug 18 19:53:29 2020 GMT]
HOST: xsjsagarwal
EXE: /opt/xilinx/xrt/bin/unwrapped/xbmgmt2
[xbmgmt] ERROR: Root privileges required
```
All xbmgmt commands should require sudo. If there is a scenario where we are able to run xbmgmt2 without sudo, we need to fix that immediately. Can you point me to a case where you are able to do so? I can take a look.

Reply:
not all xbmgmt subcommands needs sudo. xbmgmt scan is one of them.

Reply:
Oh, I see. 
I checked and we don't have sudo enforced globally in xbmgmt2. So this function is good, sorry for the confusion :)